src/pybind/mgr/orchestrator/module.py

```diff
@@ -486,10 +548,15 @@ def _apply_osd(self, all_available_devices=False, inbuf=None):
                 )
             ]
 
+        if preview:
+            # Not sure if this is too hacky..
+            _ = [setattr(dg, 'unmanaged', True) for dg in dg_specs]
```

Comment:
hm, IMO this is a bit too hacky :smile: 

When adding `--preview`, the OSDSpec should IMO **not** be added to the cephadm spec store but only print the results. wdyt?

Reply:
It's a matter of expectations and workflow I guess..

However, I'm on your side here. I'll skip adding specs to the store if the `preview` flag is passed.

Reply:
done